org.eclipse.xtext/src/org/eclipse/xtext/resource/persistence/StorageAwareResource.java

```diff
@@ -59,6 +82,17 @@ public void load(Map<?, ?> options) throws IOException {
 					eAdapters.clear();
 				}
 				unload();
+			} catch (StackOverflowError e) {
+			      trimStackOverflowErrorStackTrace(e);
+			      LOG.warn("Failed to load " + uri + " from storage", e); //$NON-NLS-1$//$NON-NLS-2$
+
+			      if (contents != null) {
```

Comment:
Nit: please extract a protected method `unload(Throwable t)`?  that's used from both catch blocks.

Reply:
Maybe call it `loadFromStorageFailed(Throwable)`

Reply:
@szarnekow , done